src/org/daisy/dotify/formatter/impl/core/HeightCalculator.java

```diff
@@ -30,11 +31,21 @@ float getRowSpacing(Row r) {
     }
 
     void addRow(Row r) {
+        /*
+            This implemetation is specific for the RowImpl. If someone would create an another
+            implementation of Row it's not likely that they would implement this function.
+         */
+        if (r instanceof RowImpl && ((RowImpl) r).isInvisible()) {
```

Comment:
Ok, I see your comment. But it breaks the way the API and the implementation are designed.
The API and the implementation are designed in such a way that casting is usually not used, instead the API is extended. You could do the same thing here and add method `isInvisible()` to the API.

@BertFrees what do you think?

Reply:
Yes I've thought about this too. For HeightCalculator it's not an issue. Actually HeightCalculator should simply accept RowImpl instead of Row. It's just an internal class used by PageImpl. There is no reason it needs to accept the more general Row.

For PEFMediaWriter it's different, because it's an implementation of PagedMediaWriter and you might want to swap it for another writer. In this case I think the code for skipping the invisible rows should move to some place else. It doesn't belong in PEFMediaWriter.

I didn't mention this however because it was a relatively small issue that can be fixed easily when a problem arises, but ideally it should be changed now, yes.

Reply:
And to answer your question: I don't think `isInvisible()` belongs in the API. It's more of an internal concept.